core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -164,6 +164,27 @@ private[spark] object TestUtils {
     createCompiledClass(className, destDir, sourceFile, classpathUrls)
   }
 
+  /** Create a dummy compile jar for a given package, classname.  Jar will be placed in destDir */
+  def createDummyJar(destDir: String, packageName: String, className: String): String = {
```

Comment:
I know this is moved code, but could we just use `createJarWithClasses` or do the R tests require these inner functions?

If we do keep this though I'd give it a clearer name and description because without reading the code, `createDummyJar` and `createJarWithClasses`  don't seem all that different.